app/services/statements/search.rb

```diff
@@ -50,6 +56,13 @@ def where_statement_date(statement_date)
       @scope = scope.with_statement_date(year:, month:)
     end
 
+    def where_deadline_date(deadline_date)
+      return if ignore?(filter: deadline_date)
+      return if deadline_date.blank?
+
+      @scope = scope.where(deadline_date: deadline_date..)
```

Comment:
Instead of making this a range here can we accept the range (or exact date) in case other callers want statements between two dates?